### PR TITLE
Create TreeSort.md

### LINKED ISSUE #323: [Feature Request] <Tree Sort>
### Is your feature request related to a problem? Please describe.
Tree Sort is a sorting technique based on Binary Search Tree (BST). I'm looking for a clean and efficient implementation that provides better performance for moderate-sized datasets compared to traditional sorting algorithms.

### Describe the solution you'd like
An implementation of the Tree Sort algorithm that inserts elements into a BST and performs an in-order traversal to retrieve a sorted list.

### Describe alternatives you've considered
Alternatives like Merge Sort and Quick Sort are efficient but do not take advantage of the sorted nature of BSTs for specific types of data.

### Additional context
Tree Sort works well when the data can be stored in a BST and performs better in average cases, but it may degrade to O(n²) in unbalanced cases. Consider using a balanced tree for optimal performance.

---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by Gopal0Gupta**
### Description
Tree Sort has been implemented. It builds a Binary Search Tree (BST) from input elements and retrieves them in sorted order via an in-order traversal. This algorithm is efficient in balanced cases but can degrade to O(n²) when the tree becomes unbalanced.

Fixes #323 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Gopal0Gupta** [comment]
@ajay-dhangar please merge it

File: docs/algorithms/sorting-algorithms/TreeSort.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+)
+O(nlogn), it can degrade to 
+𝑂
+(
+𝑛
+2
+)
+O(n 
+2
+ ) in the worst case, making it less suitable for practical use in many scenarios compared to other algorithms like QuickSort or MergeSort.
```
Comment (ajay-dhangar) [inline docs/algorithms/sorting-algorithms/TreeSort.md]: rearrange formats

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

File: docs/algorithms/sorting-algorithms/TreeSort.md (hunk 1 above)
Comment (Gopal0Gupta) [reply docs/algorithms/sorting-algorithms/TreeSort.md]: is it okk now

**Comment by Gopal0Gupta** [comment]
> update code based on suggestions

Done

**Pull request merged**